### PR TITLE
Top Level Groups Enumeration

### PR DESCRIPTION
**Pull request by some1one**
+added WhereNotNull IEnumerable Extension
+added GetTopLevelMajorRecordTypes to MajorRecordTypeEnumerator
+added to cheatsheat
*changed cache list type to immutableArray

### CODE REVIEW

**Comment by some1one** [comment]
rebased on dev and removed the doc commits from the other pull request

File: Mutagen.Bethesda.Core/Extensions/EnumerableExt.cs [hunk 1]
```diff
@@ -4,6 +4,11 @@ namespace Mutagen.Bethesda;
 
 internal static class EnumerableExt
 {
+    public static IEnumerable<T> WhereNotNull<T>(this IEnumerable<T?> enumer)
```
Comment (Noggog) [inline Mutagen.Bethesda.Core/Extensions/EnumerableExt.cs:7]: Already got an extension like this over in https://github.com/Noggog/CSharpExt/blob/89a5e0918ae8a212b592b890a5b62ca1209f2120/Noggog.CSharpExt/Extensions/EnumerableExt.cs#L257

But I do like the `Where__` naming better.   Feel free to open a PR over there, and mark the old ones obsolete

**Review by Noggog — CHANGES_REQUESTED** [review]
Rebase went a bit weird and some commits got in both, but all good!
image

Only thing I'd say is maybe (re)move the null extension, and should be good to go!

**Comment by Noggog** [comment]
Ah, i can handle the small tweaks.   Like i mentioned, gonna remove the new WhereNotNull method, but feel free to add to CSharpExt if you feel inspired to do so

**Review by Noggog — APPROVED** [review]

**Pull request merged**